src/tokens/ERC1155/presets/lootbox/ERC1155Lootbox.sol

```diff
@@ -0,0 +1,114 @@
+    function initialize(
+        address owner,
+        string memory tokenName,
+        string memory tokenBaseURI,
+        string memory tokenContractURI,
+        address royaltyReceiver,
+        uint96 royaltyFeeNumerator
+    ) public virtual override {
+        _grantRole(MINT_ADMIN_ROLE, owner);
+        super.initialize(owner, tokenName, tokenBaseURI, tokenContractURI, royaltyReceiver, royaltyFeeNumerator);
+    }
+
+    /// @inheritdoc IERC1155LootboxFunctions
+    function setBoxContent(bytes32 _merkleRoot, uint256 _boxSupply) external onlyRole(MINT_ADMIN_ROLE) {
+        merkleRoot = _merkleRoot;
+        boxSupply = _boxSupply;
+    }
+
+    /// @inheritdoc IERC1155LootboxFunctions
+    function commit() external {
+        if (_commitments[msg.sender] != 0) {
+            revert PendingReveal();
+        }
+        if (balanceOf(msg.sender, 1) == 0) {
+            revert NoBalance();
+        }
+        _burn(msg.sender, 1, 1);
+        uint256 revealAfterBlock = block.number + 1;
+        _commitments[msg.sender] = revealAfterBlock;
+
+        emit Commit(msg.sender, revealAfterBlock);
+    }
+
+    /// @inheritdoc IERC1155LootboxFunctions
+    function reveal(address user, BoxContent calldata boxContent, bytes32[] calldata proof) external {
+        uint256 revealIdx = getRevealId(user);
+        bytes32 leaf = keccak256(abi.encode(revealIdx, boxContent));
+
+        if (!MerkleProof.verify(proof, merkleRoot, leaf)) {
+            revert InvalidProof();
+        }
+
+        delete _commitments[user];
+        _claimedIdxs[revealIdx] = true;
+
+        for (uint256 i = 0; i < boxContent.tokenAddresses.length; i++) {
+            IERC1155ItemsFunctions(boxContent.tokenAddresses[i]).mint(
+                user, boxContent.tokenIds[i], boxContent.amounts[i], ""
+            );
+        }
+    }
+
+    /// @inheritdoc IERC1155LootboxFunctions
+    function refundBox(address user) external {
+        if (_commitments[user] == 0) {
+            revert NoCommit();
+        }
+        if (uint256(blockhash(_commitments[user])) != 0 || block.number <= _commitments[user]) {
+            revert PendingReveal();
+        }
+        delete _commitments[user];
+        _mint(user, 1, 1, "");
+    }
+
+    /// @inheritdoc IERC1155LootboxFunctions
+    function getRevealId(address user) public view returns (uint256 revealIdx) {
+        if (_commitments[user] == 0) {
+            revert NoCommit();
+        }
+
+        bytes32 blockHash = blockhash(_commitments[user]);
+
+        if (uint256(blockHash) == 0) {
+            revert InvalidCommit();
+        }
+
+        revealIdx = uint256(keccak256(abi.encode(blockHash, user))) % boxSupply;
+
+        uint256 iterations;
+
+        while (_claimedIdxs[revealIdx]) {
+            revealIdx++;
+            if (revealIdx >= boxSupply) revealIdx = 0;
+            iterations++;
+            if (iterations == boxSupply) {
+                revert AllBoxesOpened();
+            }
+        }
```

Comment:
If there is only a few lootboxes to open, say out of 100,000, won't this potentially create a crazy loop?